api/scripts/prod/populate-campaign-participation-id-in-knowledge-element-snapshot.js

```diff
@@ -0,0 +1,130 @@
+import { knex } from '../../db/knex-database-connection.js';
+import { CampaignTypes } from '../../src/prescription/shared/domain/constants.js';
+import { Script } from '../../src/shared/application/scripts/script.js';
+import { ScriptRunner } from '../../src/shared/application/scripts/script-runner.js';
+
+const DEFAULT_CHUNK_SIZE = 10000;
+const DEFAULT_PAUSE_DURATION = 2000;
+
+const pause = async (duration) => {
+  return new Promise((resolve) => {
+    setTimeout(resolve, duration);
+  });
+};
+
+// Définition du script
+export class PopulateCampaignParticipationIdScript extends Script {
+  constructor() {
+    super({
+      description:
+        'This script will populate the column knowledge-element-snapshots.campaignParticipationId with campaign-participations.id',
+      permanent: true,
+      options: {
+        chunkSize: {
+          type: 'number',
+          default: DEFAULT_CHUNK_SIZE,
+          description: 'number of records to update in one update',
+        },
+        pauseDuration: {
+          type: 'number',
+          default: DEFAULT_PAUSE_DURATION,
+          description: 'Time in ms between each chunk processing',
+        },
+      },
+    });
+  }
+
+  async handle({ options, logger, dependencies = { pause } }) {
+    const result = await knex('knowledge-element-snapshots').count().whereNull('campaignParticipationId').first();
+    logger.info(`Try to populate ${result.count} missing campaignParticipationId`);
```

Comment:
suggestion: en attendant qu'on ajoute cela nativement dans le script, comme évoqué pendant la MEP, vous pouvez ajouter un scope à votre logger, de sorte à retrouver facilement ces logs dans datadog.
Vous pouvez écrire
`logger.info({ event: 'PopulateCampaignParticipationIdScript' }, `Try to populate ${result.count} missing campaignParticipationId`);`
et faire de même pour tous les appels à logger.
Ou bien utiliser un childLogger comme ceci, au début du fichier vous faites :
```
import { child } from '../../../shared/infrastructure/utils/logger.js';
const logger = child('le_nom_de_votre_choix', { event: '<nom du script>' });
```
et après vous utilisez le logger comme avant sans avoir à préciser le scope via event